### ISSUE TITLE
feature request: measure Time To Load (TTL)

### ISSUE THREAD

**Issue by Krastanov** [MilesCranmer/AirspeedVelocity.jl#3]
the result of `@time @eval using PackageName` after precompilation has been ensured (probably can not be part of a BenchmarkGroup)

**Comment by Krastanov** [comment]
Yes, that should be doable.

Here is a shell script I am using for TTFX and TTL https://github.com/Krastanov/QuantumClifford.jl/blob/master/benchmark/run_ttfx_bechmarks.sh

**Comment by MilesCranmer** [comment]
Nice. Could probably just paste that first snippet in, and have a CLI flag to turn it off?

I'm not sure how to do the second snippet generally though. Perhaps there is a way to trick BenchmarkGroup to not compile each time (e.g., have it internally call `@eval` inside the macro, with a random function name?).

**Comment by MilesCranmer** [comment]
e.g., here: https://github.com/MilesCranmer/AirspeedVelocity.jl/blob/9a8915db111a9503429f856e40e463b7424dd3cd/src/Utils.jl#L115

could add: 
```julia
using BenchmarkTools: run, BenchmarkGroup, Trial
using JSON3: JSON3
using Pkg: Pkg

extra_results = Trial(
    ... # I don't know the syntax)
    ; times=[@time @eval using $(spec.name)]
)

...

results = (
    benchmarks=results,
    metadata=extra_results
)

...
```

**Comment by Krastanov** [comment]
I do not think that example would work, because only the first run of `using` will be slow (the one we want to measure). All the other ones will be fast. I think this can not be done cleanly with a `Trial` object and needs to be done as a separate custom source of benchmark data.
